### ISSUE TITLE
Centos 7.2 with Python 2.7: ImportError: No module named card.utils

### ISSUE THREAD

**Issue by martynion** [PodgroupConnectivity/kiopcgenerator#1]
Hi,

I am trying to use it on CentOS Linux release 7.2.1511 and I get the follwoing error:
[root@APPServer kiopcgenerator]# kiopcgen
Traceback (most recent call last):
  File "/usr/bin/kiopcgen", line 4, in <module>
    __import__('pkg_resources').run_script('kiopcgenerator==0.1.2', 'kiopcgen')
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 666, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1469, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/lib/python2.7/site-packages/kiopcgenerator-0.1.2-py2.7.egg/EGG-INFO/scripts/kiopcgen", line 14, in <module>

  File "build/bdist.linux-x86_64/egg/kiopcgenerator/__init__.py", line 9, in <module>
  File "build/bdist.linux-x86_64/egg/kiopcgenerator/lib.py", line 17, in <module>
ImportError: No module named card.utils

Python version is  2.7.5 and this are the logs for the instalation:
[root@APPServer kiopcgenerator]# python setup.py install
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'setup_requies'
  warnings.warn(msg)
running install
running bdist_egg
running egg_info
creating kiopcgenerator.egg-info
writing requirements to kiopcgenerator.egg-info/requires.txt
writing kiopcgenerator.egg-info/PKG-INFO
writing top-level names to kiopcgenerator.egg-info/top_level.txt
writing dependency_links to kiopcgenerator.egg-info/dependency_links.txt
writing manifest file 'kiopcgenerator.egg-info/SOURCES.txt'
reading manifest file 'kiopcgenerator.egg-info/SOURCES.txt'
writing manifest file 'kiopcgenerator.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib
creating build/lib/kiopcgenerator
copying kiopcgenerator/lib.py -> build/lib/kiopcgenerator
copying kiopcgenerator/__init__.py -> build/lib/kiopcgenerator
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/kiopcgenerator
copying build/lib/kiopcgenerator/lib.py -> build/bdist.linux-x86_64/egg/kiopcgenerator
copying build/lib/kiopcgenerator/__init__.py -> build/bdist.linux-x86_64/egg/kiopcgenerator
byte-compiling build/bdist.linux-x86_64/egg/kiopcgenerator/lib.py to lib.pyc
byte-compiling build/bdist.linux-x86_64/egg/kiopcgenerator/__init__.py to __init__.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
installing scripts to build/bdist.linux-x86_64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/scripts-2.7
copying and adjusting kiopcgen -> build/scripts-2.7
changing mode of build/scripts-2.7/kiopcgen from 644 to 755
creating build/bdist.linux-x86_64/egg/EGG-INFO/scripts
copying build/scripts-2.7/kiopcgen -> build/bdist.linux-x86_64/egg/EGG-INFO/scripts
changing mode of build/bdist.linux-x86_64/egg/EGG-INFO/scripts/kiopcgen to 755
copying kiopcgenerator.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kiopcgenerator.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kiopcgenerator.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kiopcgenerator.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kiopcgenerator.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating dist
creating 'dist/kiopcgenerator-0.1.2-py2.7.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing kiopcgenerator-0.1.2-py2.7.egg
Copying kiopcgenerator-0.1.2-py2.7.egg to /usr/lib/python2.7/site-packages
Adding kiopcgenerator 0.1.2 to easy-install.pth file
Installing kiopcgen script to /usr/bin

Installed /usr/lib/python2.7/site-packages/kiopcgenerator-0.1.2-py2.7.egg
Processing dependencies for kiopcgenerator==0.1.2
Searching for card==0.0.1
Best match: card 0.0.1
Processing card-0.0.1-py2.7.egg
card 0.0.1 is already the active version in easy-install.pth

Using /usr/lib/python2.7/site-packages/card-0.0.1-py2.7.egg
Searching for pycrypto==2.6.1
Best match: pycrypto 2.6.1
Processing pycrypto-2.6.1-py2.7-linux-x86_64.egg
pycrypto 2.6.1 is already the active version in easy-install.pth

Using /usr/lib/python2.7/site-packages/pycrypto-2.6.1-py2.7-linux-x86_64.egg
Finished processing dependencies for kiopcgenerator==0.1.2

I tried also using repository and the same issue:
[root@APPServer ~]# pip install git+https://github.com/PodgroupConnectivity/kiopcgenerator#egg=kiopcgenerator
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting kiopcgenerator
  Cloning https://github.com/PodgroupConnectivity/kiopcgenerator to /tmp/pip-install-8bSiRB/kiopcgenerator
  Running command git clone -q https://github.com/PodgroupConnectivity/kiopcgenerator /tmp/pip-install-8bSiRB/kiopcgenerator
Requirement already satisfied: pycrypto in /usr/lib/python2.7/site-packages/pycrypto-2.6.1-py2.7-linux-x86_64.egg (from kiopcgenerator) (2.6.1)
Requirement already satisfied: card@ git+https://github.com/PodgroupConnectivity/card.git@b8eb2c70eaad754a631a9fcbd1ce0dba4b58a662#egg=card-0.3.1 from git+https://github.com/PodgroupConnectivity/card.git@b8eb2c70eaad754a631a9fcbd1ce0dba4b58a662#egg=card-0.3.1 in /usr/lib/python2.7/site-packages/card-0.0.1-py2.7.egg (from kiopcgenerator) (0.0.1)
Building wheels for collected packages: kiopcgenerator
  Building wheel for kiopcgenerator (setup.py) ... done
  Created wheel for kiopcgenerator: filename=kiopcgenerator-0.1.2-py2-none-any.whl size=8247 sha256=130954e72ef2402a5958633122a0a5617cc8d053a357ff1de1a1f39fd9f4ae3a
  Stored in directory: /tmp/pip-ephem-wheel-cache-4szNSW/wheels/c3/f0/64/7c584b05b7721fc5ed9b0f5d3cc701aa6975721864264b28ca
Successfully built kiopcgenerator
Installing collected packages: kiopcgenerator
Successfully installed kiopcgenerator-0.1.2
[root@APPServer ~]# ki
kill      killall5  kiopcgen
[root@APPServer ~]# kiopcgen
Traceback (most recent call last):
  File "/usr/bin/kiopcgen", line 14, in <module>
    import kiopcgenerator
  File "/usr/lib/python2.7/site-packages/kiopcgenerator/__init__.py", line 9, in <module>
    from lib import gen_ki, gen_eki, gen_opc, gen_opc_eki
  File "/usr/lib/python2.7/site-packages/kiopcgenerator/lib.py", line 17, in <module>
    from card.utils import stringToByte, byteToString
ImportError: No module named card.utils
[root@APPServer ~]#

**Comment by mahfuz110244** [comment]
You need to setup this module in your virtual environment. All this four module need to setup properly
card==0.3.1
kiopcgenerator==0.1.2
pycrypto==2.6.1
pyscard==1.9.9

**Comment by fontanon** [comment]
Thanks @martynion @mahfuz110244 for your help detecting and resolving the issue. 

I've added a ```requirements.txt``` in the master branch to ease with the installation of dependencies. 

```
fontanon@baphomet:~/Work/kiopcgenerator$ virtualenv .env
New python executable in /home/fontanon/Work/kiopcgenerator/.env/bin/python
Installing setuptools, pip, wheel...
done.
fontanon@baphomet:~/Work/kiopcgenerator$ source .env/bin/activate
(.env) fontanon@baphomet:~/Work/kiopcgenerator$ pip install -r requirements.txt
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Requirement already satisfied: wheel in ./.env/lib/python2.7/site-packages (from -r requirements.txt (line 1)) (0.34.2)
Processing /home/fontanon/.cache/pip/wheels/87/7a/77/5823300c78a6745982cf0880e69ca878afba40d8b82c1136cf/pyscard-1.9.9-cp27-cp27mu-linux_x86_64.whl
Processing /home/fontanon/.cache/pip/wheels/27/02/5e/77a69d0c16bb63c6ed32f5386f33a2809c94bd5414a2f6c196/pycrypto-2.6.1-cp27-cp27mu-linux_x86_64.whl
Collecting card-0.3.1
  Cloning https://github.com/PodgroupConnectivity/card.git (to revision b8eb2c70eaad754a631a9fcbd1ce0dba4b58a662) to /tmp/pip-install-acCpwQ/card-0.3.1
  Running command git clone -q https://github.com/PodgroupConnectivity/card.git /tmp/pip-install-acCpwQ/card-0.3.1
  WARNING: Generating metadata for package card-0.3.1 produced metadata for project name card. Fix your #egg=card-0.3.1 fragments.
Building wheels for collected packages: card, card
  Building wheel for card (setup.py) ... done
  Created wheel for card: filename=card-0.3.1-py2-none-any.whl size=57686 sha256=6b5c69df9edface76832e16142b8a40ae8706410968af97936d83f5eda550789
  Stored in directory: /home/fontanon/.cache/pip/wheels/5f/ca/26/523d8f517eac16170bada0b32dcf119d001da9316d6bfc8be8
  Building wheel for card (setup.py) ... done
  Created wheel for card: filename=card-0.3.1-py2-none-any.whl size=57686 sha256=927b7616d12cd82cfb30586406dfa5f1cde9317b60bf225db4ab06783936518a
  Stored in directory: /tmp/pip-ephem-wheel-cache-VtWOTL/wheels/43/0b/ea/d7de5a2795ed0127241b6833e81b7112b62e24d6a1e0667a14
Successfully built card card
Installing collected packages: pyscard, pycrypto, card
Successfully installed card-0.3.1 pycrypto-2.6.1 pyscard-1.9.9
(.env) fontanon@baphomet:~/Work/kiopcgenerator$ python setup.py install
running install
running bdist_egg
running egg_info
creating kiopcgenerator.egg-info
writing kiopcgenerator.egg-info/PKG-INFO
writing top-level names to kiopcgenerator.egg-info/top_level.txt
writing dependency_links to kiopcgenerator.egg-info/dependency_links.txt
writing manifest file 'kiopcgenerator.egg-info/SOURCES.txt'
reading manifest file 'kiopcgenerator.egg-info/SOURCES.txt'
writing manifest file 'kiopcgenerator.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/kiopcgenerator
copying kiopcgenerator/lib.py -> build/lib.linux-x86_64-2.7/kiopcgenerator
copying kiopcgenerator/__init__.py -> build/lib.linux-x86_64-2.7/kiopcgenerator
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/kiopcgenerator
copying build/lib.linux-x86_64-2.7/kiopcgenerator/lib.py -> build/bdist.linux-x86_64/egg/kiopcgenerator
copying build/lib.linux-x86_64-2.7/kiopcgenerator/__init__.py -> build/bdist.linux-x86_64/egg/kiopcgenerator
byte-compiling build/bdist.linux-x86_64/egg/kiopcgenerator/lib.py to lib.pyc
byte-compiling build/bdist.linux-x86_64/egg/kiopcgenerator/__init__.py to __init__.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
installing scripts to build/bdist.linux-x86_64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/scripts-2.7
copying and adjusting kiopcgen -> build/scripts-2.7
changing mode of build/scripts-2.7/kiopcgen from 664 to 775
creating build/bdist.linux-x86_64/egg/EGG-INFO/scripts
copying build/scripts-2.7/kiopcgen -> build/bdist.linux-x86_64/egg/EGG-INFO/scripts
changing mode of build/bdist.linux-x86_64/egg/EGG-INFO/scripts/kiopcgen to 775
copying kiopcgenerator.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kiopcgenerator.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kiopcgenerator.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying kiopcgenerator.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating dist
creating 'dist/kiopcgenerator-0.1.4-py2.7.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing kiopcgenerator-0.1.4-py2.7.egg
Copying kiopcgenerator-0.1.4-py2.7.egg to /home/fontanon/Work/kiopcgenerator/.env/lib/python2.7/site-packages
Adding kiopcgenerator 0.1.4 to easy-install.pth file
Installing kiopcgen script to /home/fontanon/Work/kiopcgenerator/.env/bin

Installed /home/fontanon/Work/kiopcgenerator/.env/lib/python2.7/site-packages/kiopcgenerator-0.1.4-py2.7.egg
Processing dependencies for kiopcgenerator==0.1.4
Finished processing dependencies for kiopcgenerator==0.1.4
(.env) fontanon@baphomet:~/Work/kiopcgenerator$ kiopcgen -o d34e4bdb8f6a473caff6cbb3c9b22bcf -t dd48b648a4a614d7eb457cd08c30121b
{'KI': '8C4617AD367B4F129788F36CF5DD6CC7',
 'OPC': '2BCFB1C180FF72E854D682716A90C254',
 'eKI': '2876B151B02389D7C32E09DBD6D6FEE3'}
```
